nanoFramework.Hardware.Esp32.Rmt/RmtChannelSettings.cs

```diff
@@ -0,0 +1,130 @@
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// Base class for shared RMT Channel settings.
+    /// </summary>
+    public abstract class RmtChannelSettings
+    {
+        private int _channel;
+        private int _pinNumber;
+        private byte _clockDivider;
+        private byte _numberOfMemoryBlocks;
+        private int _bufferSize;
+        private bool _signalInverterEnabled;
+
+        /// <summary>
+        /// Gets or sets the channel number.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int Channel
+        {
+            get => _channel;
+            set => _channel = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the GPIO pin number to be used with the specified channel.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int PinNumber
+        {
+            get => _pinNumber;
+            set => _pinNumber = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the clock divider value.
+        /// </summary>
+        /// <remarks>
+        /// ESP IDF v4.4.3 only supports the APB bus clock @ 80Mhz.
+        /// The default value for the divider is 80 (80Mhz / 80 = 1Mhz = 1us ticks).
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public byte ClockDivider
+        {
+            get => _clockDivider;
+            set => _clockDivider = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the number of memory blocks available to the current channel.
+        /// </summary>
+        /// <remarks>
+        /// This function is used to configure the amount of memory blocks allocated to a channel.
+        /// The 8 channels share a 512x32-bit RAM block which can be read and written by the processor cores over the APB bus,
+        /// as well as read by the transmitters and written by the receivers.
+        /// This means that every channel gets a single memory block equaling 64x32-bit.
+        /// If memory block number of one channel is set to a value greater than 1, this channel will occupy the memory block of the next channel.
+        /// Channel 0 can use at most 8 blocks of memory, accordingly channel 7 can only use one memory block.
+        /// </remarks>
+        public byte NumberOfMemoryBlocks { get => _numberOfMemoryBlocks; set => _numberOfMemoryBlocks = value; }
```

Comment:
This (and following ones) shouldn't be inlined. Hit CTRL+K CTRL+D to have VS take care of the basic formatting for you.